packages/react-components/react-drawer/src/components/InlineDrawer/useInlineDrawer.ts

```diff
@@ -14,22 +14,20 @@ import { useDrawerDefaultProps } from '../../shared/useDrawerDefaultProps';
  * @param props - props from this instance of InlineDrawer
  * @param ref - reference to root HTMLElement of InlineDrawer
  */
-export const useInlineDrawer_unstable = (
-  props: InlineDrawerProps,
-  ref: React.Ref<HTMLDivElement>,
-): InlineDrawerState => {
+export const useInlineDrawer_unstable = (props: InlineDrawerProps, ref: React.Ref<HTMLElement>): InlineDrawerState => {
   const { size, position, open } = useDrawerDefaultProps(props);
   const { separator = false } = props;
 
-  const motion = useMotion<HTMLDivElement>(open);
+  const motion = useMotion<HTMLElement>(open);
+  const defaultTag = props.as || 'div';
 
   return {
     components: {
-      root: 'div',
+      root: defaultTag,
     },
 
     root: slot.always(
-      getIntrinsicElementProps('div', {
+      getIntrinsicElementProps(defaultTag, {
```

Comment:
`getIntrinsicElementProps` will already opt-in into the `props.as` property instead of the tag, this might not be necessary

Reply:
https://github.com/microsoft/fluentui/blob/e46ffef61107d0b56e7644152e1fbfdff6adbeb5/packages/react-components/react-utilities/src/compose/getIntrinsicElementProps.ts#L15-L30